lib/routes/user/embedded_widgets/bitrefill_widget.dart

```diff
@@ -0,0 +1,83 @@
+import 'dart:convert' as JSON;
+import 'package:flutter/material.dart';
+import 'package:breez/bloc/blocs_provider.dart';
+import 'package:breez/bloc/invoice/invoice_bloc.dart';
+import 'package:flutter_webview_plugin/flutter_webview_plugin.dart';
+import 'package:breez/theme_data.dart' as theme;
+import 'package:breez/widgets/back_button.dart' as backBtn;
+
+class BitrefillPage extends StatefulWidget {
+  final String _title = "Bitrefill";
+
+  @override
+  State<StatefulWidget> createState() {
+    return new BitrefillPageState();
+  }
+}
+
+class BitrefillPageState extends State<BitrefillPage> {
+  final _widgetWebview = new FlutterWebviewPlugin();
+  String _lightningLink;
+
+  @override
+  void initState() {
+    super.initState();
+    _lightningLink = null;
+
+    _widgetWebview.lightningLinkStream.listen((value) {
+      final order = JSON.jsonDecode(value);
+      setState(() {
+        _lightningLink = order['uri'];
+        _widgetWebview.hide();
+        Navigator.of(context).pushNamed('/home');
+      });
+    });
+
+    _widgetWebview.onStateChanged.listen((state) async {
+      if (state.type == WebViewState.finishLoad) {
+        String script =
+            'window.addEventListener("message", receiveMessage, false);' +
+                'function receiveMessage(event) {Android.getPostMessage(event.data);}';
+        _widgetWebview.evalJavascript(script);
+      }
+    });
+
+    _widgetWebview.onDestroy.listen((_) {
+      if (Navigator.canPop(context)) {
+        Navigator.of(context).pop();
+      }
+    });
+  }
+
+  @override
+  void dispose() {
+    // TODO: implement dispose
+    super.dispose();
+    _widgetWebview.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    InvoiceBloc invoiceBloc = AppBlocsProvider.of<InvoiceBloc>(context);
+    if (_lightningLink != null) {
+      invoiceBloc.newLightningLinkSink.add(_lightningLink);
```

Comment:
triggering events (including adding to sink) from the build method is not the way to do it. build method is only for rendering. Instead it is better to push the link to the sink at the same moment we get it from the webview  lightningLinkStream.